### PR TITLE
7160 implement pagination

### PR DESCRIPTION
**Pull request by weskubo-cgi**

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java [hunk 1]
```diff
@@ -60,6 +68,20 @@ public class AuditService {
...
 	private OrganizationRepository organizationRepository;
+	
+	/** Maps simple sort names to their JPA equivalent */
+	private static Map<String, String> sortMap = new HashMap<>();
+	
+	static {
+		sortMap.put("affectedPartyId", "identifier");
+		sortMap.put("affectedPartyType", "identifierType");
+		sortMap.put("organization", "transaction.organization");
+		sortMap.put("organization", "transaction.organization");
```
Comment (AnumehaSrivastava05) [inline backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java:79]: Added twice

Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java]: Fixed.

File: frontend/src/views/DataTableDemo.vue [hunk 2]
```diff
@@ -0,0 +1,105 @@
+<template>
```
Comment (daveb-hni) [inline frontend/src/views/DataTableDemo.vue:1]: Have we got a task or TODO to remove this once it's not needed as a reference?

**Review by daveb-hni — COMMENTED** [review]
Please see comments

File: frontend/src/views/DataTableDemo.vue (hunk 2 above)
Comment (weskubo-cgi) [reply frontend/src/views/DataTableDemo.vue:1]: I will remove this.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java [hunk 3]
```diff
@@ -205,15 +228,24 @@ public List<Organization> getOrganizations() {
...
 	 * @return
 	 */
-	public List<AffectedParty> getAffectedParties(List<String> types, List<String> organizations, String userId, LocalDate startDate,
-			LocalDate endDate) {
+	public Page<AffectedParty> getAffectedParties(List<String> types, List<String> organizations, String userId, LocalDate startDate,
+			LocalDate endDate, int page, int rows, String sortField, String sortDirection) {
+		logger.info("Querying page {} with {} rows", page, rows);
 		try {
 			Date formattedStartDate = convertLocalDateToDate(startDate);
-			Date formattedendDate = convertLocalDateToDate(endDate);
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java:212]: Fixed.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java [hunk 4]
```diff
@@ -205,15 +228,24 @@ public List<Organization> getOrganizations() {
...
+			Date formattedendDate = convertLocalDateToDate(endDate);			
+
+			String property = sortMap.get(sortField);
+			if (StringUtils.isBlank(property)) {
+				property = DEFAULT_SORT;
+			}
+			Direction direction = StringUtils.isNotEmpty(sortDirection) ? Direction.valueOf(sortDirection) : Direction.DESC;
+			Sort sort = Sort.by(direction, property);
+			
+			Pageable pageable = PageRequest.of(page, rows, sort);
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java:245]: It's only used once and the logic is currently specific to this query so I will leave as is.

**Review by daveb-hni — APPROVED** [review]
Looks good

**Pull request merged**